packages/arch3-core/src/archwayclient.ts

```diff
@@ -26,26 +26,26 @@ import {
 export class ArchwayClient extends CosmWasmClient implements IArchwayQueryClient {
   private readonly archwayQueryClient: IArchwayQueryClient;
 
-  protected constructor(tmClient: TendermintClient | undefined) {
-    super(tmClient);
-    this.archwayQueryClient = createArchwayQueryClient(tmClient);
+  protected constructor(cometClient: CometClient | undefined) {
+    super(cometClient);
+    this.archwayQueryClient = createArchwayQueryClient(cometClient);
   }
 
   /**
-   * Creates an instance by connecting to the given Tendermint RPC endpoint.
+   * Creates an instance by connecting to the given Tendermint/Comet RPC endpoint.
    *
    * @param endpoint - String URL of the RPC endpoint to connect or an {@link HttpEndpoint} object.
    * @returns An {@link ArchwayClient} connected to the endpoint.
    *
-   * @see Use {@link ArchwayClient.create} if you need Tendermint 0.37 support.
+   * @see Use {@link ArchwayClient.create} if you need Comet 0.38 support.
    */
   public static override async connect(endpoint: string | HttpEndpoint): Promise<ArchwayClient> {
     const tmClient = await Tendermint37Client.connect(endpoint);
```

Comment:
Please use the function `connectComet` available in the `tendermint-rpc` package:

```suggestion
    const cometClient = await connectComet(endpoint);
```

This function queries the chain and selects the right client version based on the Tendermint/CometBFT version running, making it backward compatible between the current version running on `constantine` and `mainnet`:

```
function connectComet(endpoint: string | HttpEndpoint): Promise<CometClient>
Auto-detects the version of the backend and uses a suitable client.
```